drivers/input/input_cf1133.c

```diff
@@ -0,0 +1,358 @@
+		input_report_abs(dev, INPUT_ABS_Y, y, false, K_FOREVER);
+		LOG_DBG("Finger keeps touching x = %i y = %i", x, y);
+	}
+
+	data->pressed_old = data->pressed;
+
+	return 0;
+}
+
+static void cf1133_work_handler(struct k_work *work)
+{
+	struct cf1133_data *data = CONTAINER_OF(work, struct cf1133_data, work);
+
+	cf1133_process(data->dev);
+}
+
+#ifdef CONFIG_INPUT_CF1133_INTERRUPT
+
+static void cf1133_isr_handler(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	struct cf1133_data *data = CONTAINER_OF(cb, struct cf1133_data, int_gpio_cb);
+
+	k_work_submit(&data->work);
+}
+#else
+static void cf1133_timer_handler(struct k_timer *timer)
+{
+	struct cf1133_data *data = CONTAINER_OF(timer, struct cf1133_data, timer);
+
+	k_work_submit(&data->work);
+}
+#endif
+
+static int cf1133_init(const struct device *dev)
+{
+	const struct cf1133_config *config = dev->config;
+	struct cf1133_data *data = dev->data;
+	int r;
+
+	if (!device_is_ready(config->bus.bus)) {
+		LOG_ERR("I2C controller device not ready");
+		return -ENODEV;
+	}
+
+	data->dev = dev;
+	k_work_init(&data->work, cf1133_work_handler);
+
+#ifdef CONFIG_INPUT_CF1133_INTERRUPT
+
+	LOG_DBG("Int conf for TS gpio : %d", &config->int_gpio);
+
+	if (!gpio_is_ready_dt(&config->int_gpio)) {
+		LOG_ERR("Interrupt GPIO controller device not ready");
+		return -ENODEV;
+	}
+
+	r = gpio_pin_configure_dt(&config->int_gpio, GPIO_INPUT);
+	if (r < 0) {
+		LOG_ERR("Could not configure interrupt GPIO pin");
+		return r;
+	}
+
+	gpio_init_callback(&(data->int_gpio_cb), cf1133_isr_handler, BIT(config->int_gpio.pin));
+
+	r = gpio_add_callback(config->int_gpio.port, &data->int_gpio_cb);
+	if (r < 0) {
+		LOG_ERR("Could not set gpio callback");
+		return r;
+	}
+
+	r = gpio_pin_interrupt_configure_dt(&config->int_gpio, GPIO_INT_EDGE_TO_ACTIVE);
+	if (r < 0) {
+		LOG_ERR("Could not configure interrupt GPIO interrupt.");
+		return r;
+	}
+#else
+	LOG_DBG("Timer Mode");
+	k_timer_init(&data->timer, cf1133_timer_handler, NULL);
+	k_timer_start(&data->timer, K_MSEC(CONFIG_INPUT_CF1133_PERIOD),
+		      K_MSEC(CONFIG_INPUT_CF1133_PERIOD));
+#endif
+
+	r = cf1133_ts_init(data, config);
+	if (r < 0) {
+		LOG_ERR("Init information of sensor failed: %d", r);
+		return r;
+	}
+	return 0;
+}
+
+#define CF1133_INIT(index)                                                                         \
+	static const struct cf1133_config cf1133_config_##index = {                                \
+		.bus = I2C_DT_SPEC_INST_GET(index),                                                \
+		IF_ENABLED(CONFIG_INPUT_CF1133_INTERRUPT, (                                         \
+		.int_gpio = GPIO_DT_SPEC_INST_GET(index, int_gpios),
+		))
+		}
+		;
+		static struct cf1133_data cf1133_data_##index;
+		DEVICE_DT_INST_DEFINE(index, cf1133_init, NULL, &cf1133_data_##index,
+				      &cf1133_config_##index, POST_KERNEL,
+				      CONFIG_INPUT_INIT_PRIORITY, NULL);
+
+		DT_INST_FOREACH_STATUS_OKAY(CF1133_INIT)
```

Comment:
```suggestion
                };
        static struct cf1133_data cf1133_data_##index;
        DEVICE_DT_INST_DEFINE(index, cf1133_init, NULL, &cf1133_data_##index,
                              &cf1133_config_##index, POST_KERNEL,
                              CONFIG_INPUT_INIT_PRIORITY, NULL);

DT_INST_FOREACH_STATUS_OKAY(CF1133_INIT);
```

Also mind the trailing `\`, keep them aligned

Reply:
Done